platform/src/server/http/router/router.ts

```diff
@@ -128,34 +139,30 @@ export class Router {
       kibanaRequest = KibanaRequest.from(request, routeSchemas);
     } catch (e) {
       // TODO Handle failed validation
-
-      response.status(400);
-      response.json({ error: e.message });
-      return;
+      return responseToolkit.response({ error: e.message }).code(400);
     }
 
     try {
       const kibanaResponse = await handler(kibanaRequest, responseFactory);
 
       if (kibanaResponse instanceof KibanaResponse) {
```

Comment:
I think we should throw an error if the handler doesn't return a `KibanaResponse` or, perhaps, `undefined`. Instead we are allowing handlers to return anything that `responseToolkit.response()` supports, which is not defined [here](https://hapijs.com/api/#-hresponsevalue) and as far as I know supports everything from `Stream` instances to plain strings, numbers or objects. Not limiting the supported return values to a very specific list of things opens up the "supported" API too much for my taste.

Reply:
Even though return type is limited by `KibanaResponse<any> | { [key: string]: any }`, I agree, it can be a lot of different things. I think it was initially designed to short-circuit `{...} ---> HTTP 200/JSON` path. I'll limit response to `KibanaResponse` only and we'll expand it if we need to in the future.

Reply:
Yeah, as long as `KibanaResponse` is the one defining the API I'm good with it.